### ISSUE TITLE
Breaks when the server is hibernating.

### ISSUE THREAD

**Issue by Synkstar** [storm37000/gmsv_antifreeze#2]
If you join the server then leave when sv_hibernate_think is off it detects it as being frozen

**Comment by storm37000** [comment]
That can't be fixed as there is no way to safely get game data like player count inside another thread as far as I know.  I use it as a feature to restart my server once everyone is off so that workshop addons get updated.

**Comment by Synkstar** [comment]
Honestly how id do crash detection before is by having an external progem that makes a2s_info requests to the server and if its frozen then it wouldnt respond to those anymore. But yeah that does makes sense because you cant really get the player count for example from c++ through the lua api when its hibernating.

**Comment by storm37000** [comment]
This was meant to be simpler and more reliable than having to deal with external processes and trying to get them to kill the correct process on windows and linux, especially when running multiple servers on the same OS.  Also some people may have a2s proxies that cache and still respond even if the server is frozen.

**Comment by Synkstar** [comment]
Yeah true but usually a2s_info caching is done by an upstream network or another program but you could have it send directly to the server its self technically. Technically couldnt you find the player value somewhere in memory though and grab it in this case ? Like you know using an offset kind of how a game cheat would access functions or data.

**Comment by Synkstar** [comment]
https://github.com/danielga/sourcesdk-minimal/blob/cab3e07edc4a41e7e69ea645ea51c1e5c5d1be71/game/server/gameinterface.h#L127 There is a bool called m_bIsHibernating in the sourcesdk so maybe you can access that to check if its hibernating

**Comment by storm37000** [comment]
I've never worked with game cheats/raw offsets or the raw source SDK. I am thinking of adding a lua function into the module where the watchdog can be (un)paused, and you could use the https://wiki.facepunch.com/gmod/GM:GetGameDescription hook which runs once a second even when hibernated to check player count and pause/unpause it.

**Comment by storm37000** [comment]
But for now I will try importing the sourceSDK headers and see if those work.  I would make the pause on hibernate a setting you can turn on.

**Comment by storm37000** [comment]
> https://github.com/danielga/sourcesdk-minimal/blob/cab3e07edc4a41e7e69ea645ea51c1e5c5d1be71/game/server/gameinterface.h#L127 There is a bool called m_bIsHibernating in the sourcesdk so maybe you can access that to check if its hibernating

I dont really have a clue how to access that.  Its inside a class that I have to somehow get the object reference of from the game.

**Comment by wrefgtzweve** [comment]
Had this issue today, fix i wrote for it:

```lua
function timer.Create( name, delay, repetitions, func )
    if name == "antifreeze_watchdog" then
        hook.Add( "GetGameDescription", "AntifreezePing", func )
        return
    end

    return timer_Create( name, delay, repetitions, func )
end

require( "antifreeze" )


timer.Create = timer_Create
```
